src/Symfony/Bundle/FrameworkBundle/Command/DotenvDebugCommand.php

```diff
@@ -0,0 +1,139 @@
+        $this->projectDirectory = $projectDirectory;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Dotenv Variables & Files');
+
+        $envFiles = $this->getEnvFiles();
+        $availableFiles = array_filter($envFiles, function (string $file) {
+            return is_file($this->getFilePath($file));
+        });
+
+        if (\in_array('.env.local.php', $availableFiles, true)) {
+            $io->warning('Due to existing dump file (.env.local.php) all other dotenv files are skipped.');
+        }
+
+        $io->section('Scanned Files (in descending priority)');
+        $io->listing(array_map(static function (string $envFile) use ($availableFiles) {
+            return \in_array($envFile, $availableFiles, true)
+                ? sprintf('<fg=green>✓</> %s', $envFile)
+                : sprintf('<fg=red>⨯</> %s', $envFile);
+        }, $envFiles));
+
+        $io->section('Variables');
+        $io->table(
+            array_merge(['Variable', 'Value'], $availableFiles),
+            $this->getVariables($availableFiles)
+        );
+
+        $io->comment('Note real values might be different between web and CLI.');
+
+        return 0;
+    }
+
+    private function getVariables(array $envFiles): array
+    {
+        if (!\array_key_exists('SYMFONY_DOTENV_VARS', $_SERVER)) {
+            throw new \RuntimeException('Symfony Dotenv not initialized.');
+        }
+
+        $vars = explode(',', $_SERVER['SYMFONY_DOTENV_VARS'] ?? '');
+        sort($vars);
+
+        $output = [];
+        $fileValues = [];
+        foreach ($vars as $var) {
+            $realValue = $_SERVER[$var];
+            $varDetails = [$var, $realValue];
+            foreach ($envFiles as $envFile) {
+                $file = $this->getFilePath($envFile);
+                if (!is_file($file)) {
+                    continue;
+                }
+                $values = $fileValues[$envFile] ?? $fileValues[$envFile] = $this->loadValues($file);
+
+                $varString = $values[$var] ?? '<fg=yellow>n/a</>';
+                $shortenedVar = $this->getHelper('formatter')->truncate($varString, 30);
+                $varDetails[] = $varString === $realValue ? '<fg=green>'.$shortenedVar.'</>' : $shortenedVar;
+            }
+
+            $output[] = $varDetails;
+        }
+
+        return $output;
+    }
+
+    private function getEnvFiles(): array
+    {
+        $files = [
+            '.env.local.php',
+            sprintf('.env.%s.local', $this->kernelEnvironment),
+            sprintf('.env.%s', $this->kernelEnvironment),
+        ];
+
+        if ('test' !== $this->kernelEnvironment) {
+            $files[] = '.env.local';
+        }
+
+        $files[] = '.env';
+
+        if (!is_file($this->getFilePath('.env'))) {
+            $files[] = '.env.dist';
+        }
```

Comment:
im wondering if the file list should be conditional :/ given we also show checkmarks above, perhaps listing all files consistently is less confusing ... WDYT?

Reply:
well, the idea was only potential files and `.env.dist` won't get scanned, even though it exists.
we'd need a third state ... :thinking:

Reply:
further discussion resulted in:
* if .env exists => display `✓ .env`
* if .env and .env.dist exist => raise warning that .env.dist is ignored + display `✓ .env`
* if .env doesn't exist, but .env.dist does => display `✓ .env.dist`
* if .env doesn't exist and .env.dist does not as well => display `⨯ .env`